cumulusci/tasks/salesforce/tests/test_network_member_group.py

```diff
@@ -0,0 +1,594 @@
+            context.exception.args[0],
+        )
+
+        task.logger.info.assert_not_called()
+
+        task.logger.warn.assert_not_called()
+
+        task.sf.NetworkMemberGroup.create.assert_called_once_with(
+            {"NetworkId": task._network_id, "ParentId": parent_id}
+        )
+
+    def test_run_task__none_profile_names_and_permission_set_names(
+        self,
+    ):
+        network_name = "network_name"
+
+        task = create_task(CreateNetworkMemberGroups, {"network_name": network_name})
+
+        self.assertTrue(task.options.get("profile_names") is None)
+        self.assertTrue(task.options.get("permission_set_names") is None)
+
+        task._get_network_id = Mock(return_value="network_id")
+
+        task._get_network_member_group_parent_ids = Mock(return_value=set(["Id_1"]))
+
+        task._process_parent = Mock()
+        expected_process_parent_calls = [
+            call("Profile", []),
+            call("PermissionSet", []),
+        ]
+
+        # Execute the test.
+        task._run_task()
+
+        # Assert scenario execute as expected.
+        task._get_network_id.assert_called_once_with(task.options.get("network_name"))
+
+        task._get_network_member_group_parent_ids.assert_called_once_with(
+            task._get_network_id.return_value
+        )
+
+        task._process_parent.assert_has_calls(expected_process_parent_calls)
+
+    def test_run_task__string_profile_names_and_permission_set_names(
+        self,
+    ):
+        network_name = "network_name"
+        profile_names = "profile_name"
+        permission_set_names = "permission_set_name"
+
+        task = create_task(
+            CreateNetworkMemberGroups,
+            {
+                "network_name": network_name,
+                "profile_names": profile_names,
+                "permission_set_names": permission_set_names,
+            },
+        )
+
+        self.assertEqual(profile_names, task.options.get("profile_names"))
+        self.assertEqual(permission_set_names, task.options.get("permission_set_names"))
+
+        task._get_network_id = Mock(return_value="network_id")
+
+        task._get_network_member_group_parent_ids = Mock(return_value=set(["Id_1"]))
+
+        task._process_parent = Mock()
+        expected_process_parent_calls = [
+            call("Profile", [profile_names]),
+            call("PermissionSet", [permission_set_names]),
+        ]
+
+        # Execute the test.
+        task._run_task()
+
+        # Assert scenario execute as expected.
+        task._get_network_id.assert_called_once_with(task.options.get("network_name"))
+
+        task._get_network_member_group_parent_ids.assert_called_once_with(
+            task._get_network_id.return_value
+        )
+
+        task._process_parent.assert_has_calls(expected_process_parent_calls)
+
+    def test_run_task__list_profile_names_and_permission_set_names(
```

Comment:
Because you're using CumulusCI's built-in `process_list_arg()`, I don't think it's required to test each variant of input data type - that's not testing _your_ code here.

Reply:
Do you think that means we can remove the `test_run_task__list_profile_names_and_permission_set_names` ?

Reply:
Yeah, I think it's fine to remove this unit test.